scripts/replay-tx.js

```diff
@@ -1,31 +1,97 @@
+ */
 require("colors");
 const { Command } = require("commander");
 const log = require("single-line-log").stdout;
 const dotenv = require("dotenv").config();
 const { Contract, Wallet, utils } = require("ethers");
 const { getContractInterface } = require("@eth-optimism/contracts");
-const { JsonRpcProvider, getDefaultProvider } = require("@ethersproject/providers");
+const { JsonRpcProvider, InfuraProvider } = require("@ethersproject/providers");
 const program = new Command();
 
-program.requiredOption("-h, --hash <number>", "specify transaction hash");
 program.requiredOption("-n, --network <string>", "specify network");
-program.option("-k, --key <number>", "specify private key");
+program.requiredOption("-k, --key <number>", "specify private key");
+program.option("-h, --hash <number>", "specify transaction hash");
+program.option("-from, --fromBlock <number>", "specify a block number to start from");
+program.option("-to, --toBlock <number>", "specify a block number to end at");
+program.option("-c, --clearPendingTx", "clear a pending transaction");
+
 program.parse(process.argv);
 const argOptions = program.opts();
 
-const mainnetProvider = getDefaultProvider(argOptions.network);
+if (argOptions.network !== "mainnet" && argOptions.network !== "kovan") {
+  console.error("Network must be 'mainnet' or 'kovan'");
+  process.exit();
+}
+
+if (!argOptions.hash && !argOptions.fromBlock) {
+  console.error("Must provide a hash or block number to start from.");
+  return;
+}
+const provider = new InfuraProvider(argOptions.network, process.env.INFURA_KEY);
 
-const L1_MESSENGER_PROXY = "0xD1EC7d40CCd01EB7A305b94cBa8AB6D17f6a9eFE";
+// Addresses from May 11 regenesis
+const MAINNET_L1_MESSENGER_PROXY = "0x902e5fF5A99C4eC1C21bbab089fdabE32EF0A5DF";
+const KOVAN_L1_MESSENGER_PROXY = "0x78b88FD62FBdBf67b9C5C6528CF84E9d30BB28e0";
 
-const wallet = new Wallet(argOptions.key, mainnetProvider);
-const proxyL1Messenger = new Contract(L1_MESSENGER_PROXY, getContractInterface("OVM_L1CrossDomainMessenger"), wallet);
+const messengerAddress = argOptions.network === "mainnet" ? MAINNET_L1_MESSENGER_PROXY : KOVAN_L1_MESSENGER_PROXY;
+
+const wallet = new Wallet(argOptions.key, provider);
+const proxyL1Messenger = new Contract(messengerAddress, getContractInterface("OVM_L1CrossDomainMessenger"), wallet);
 
 const main = async () => {
-  const receipt = await mainnetProvider.getTransactionReceipt(argOptions.hash);
+  if (argOptions.clearPendingTx) {
+    latestTxCount = await wallet.getTransactionCount("latest");
+    pendingTxCount = await wallet.getTransactionCount("pending");
+    if (latestTxCount < pendingTxCount) {
+      console.log("Detected a pending transaction! Clearing it...");
+      await wallet.sendTransaction({
+        to: await wallet.getAddress(),
+        value: 0,
+        nonce: latestTxCount,
+      });
+    }
+    if (latestTxCount + 1 < pendingTxCount) {
```

Comment:
Do you know if there's any way to clear out /all/ pending transactions at the same time?